filebeat/harvester/reader/line.go

```diff
@@ -13,7 +13,7 @@ import (
 // using the configured codec. The reader keeps track of bytes consumed
 // from raw input stream for every decoded line.
 type Line struct {
-	rawInput   io.Reader
+	rawInput   io.ReadCloser
```

Comment:
it's more of an optional change. users can always opt to wrap io.Reader with ioutil.NopClosed if required.